### ISSUE TITLE
Accord > v0.19 not compatible with NW.js

### ISSUE THREAD

**Issue by matthew-dean** [jescalan/accord#183]
Something changed after version 0.19 of accord where it fails to resolve the path of compilers in an NW.js (http://nwjs.io/) app. (Tested 0.22 and 0.24, which failed. 0.19 succeeded.) NW.js has an otherwise normal Node.js instance, and has no problem requiring other modules, but Accord fails. Looking at the code, it looks like the `load` method changed a bit to something more complex, but not sure if that is the source of the breakage.

**Comment by jescalan** [comment]
Not sure either, and I do not use or test with NW.js. Could you dig a little deeper here?

**Comment by matthew-dean** [comment]
Well... if I change something to get it to work, it would help to know why the loading method for `adapters/` changed in the first place between those releases? What about it was not working? What was added?

**Comment by jescalan** [comment]
There are lots of places where you can get more details on what happened with a release, or between two releases. The first spot to check is the release notes. You can see the release notes for `0.20.0` [here](). You can also check out the commit log to see a summary of changes between the two versions. This page lists out what happened between 19 and 20, specifically: https://github.com/jescalan/accord/commits/master?page=5

I'll give you a personal summary as well -- there were a lot of changes between these two versions, which is why it was a breaking bump on semver. The primary change was support for multiple versions of adapters based on the version number in the `package.json` file. This was the reason for most of the changes to the `load` method. You can see that we track adapters by the semver range that they fit into now ([example](https://github.com/jescalan/accord/tree/master/lib/adapters/scss)). This way, if there is a breaking change pushed to one of the underlying compiler APIs, we can adjust without having to push a breaking release of accord. This was an extremely important change for us in order to make sure that we could promise support for compilers over the long term.

Luckily, we have a pretty extensive test suite that covers accord. So if you make a PR that breaks things, chances are the test will tell you this. I think for this type of issue, you'll just have to dive in. I'm happy to help, but this is a problem that does not, nor ever well affect me, so I'm just giving up free time to provide support for you. I do not have time to investigate and fix this on my own, you need to lead the charge, or just pin to `0.19.x`.

For getting the best support, you have to be specific with what I can provide for you. What is the actual error that is reported by your app? Have you looked for the place in the code where the error tracks back to? Have you cloned it down and tried putting in some `console.log`s to figure out what's going wrong? The more investigation you can do into this issue other than "it's broken", the more helpful I can be.

**Comment by davej** [comment]
FYI, working fine with Electron. I explicitly pass accord the full path to the `node_modules` dir (because I had issues before), perhaps try that?

```js
accord.load('coffee', '/path/to/node/modules');
```

**Comment by matthew-dean** [comment]
@davej No matter what path I pass in here, I get "undefined is not a function" when it tries to call the isFile() method on the stat object in the resolve library. I'm not sure what path it's expecting, but it doesn't seem to work.

With no path, I get: "'tried to require: less'. None found. Make sure one has been installed!"

Even if Less is installed as a dependency in the accord module AND installed in the root node_modules folder, it can't find it. However the resolving method was rewritten after 0.19, it seems it doesn't work. I'll keep trying to hack at it though.

**Comment by matthew-dean** [comment]
Hey, everyone, MY BAD.

Through debugging, I was able to determine that the issue was in the `resolve` module, which was throwing an exception when it called `stat.isFile()`. But it was doing so because when it called `statSync` in the same file, no exception was thrown, nor any stat returned.

After scratching my head for a moment about how that could be, I realized that accord was running in a child process where the `fs` module was monkey patched. Lo and behold, statSync was catching errors but not throwing them.

Apparently, this wasn't a problem in earlier versions of accord, but technically, this is a problem I caused. Sorry for the confusion!
